widgetastic_manageiq/__init__.py

```diff
@@ -1855,29 +1855,266 @@ def is_displayed(self):
+    """ Advanced Search View """
+    import expression_editor as exp_editor
+    search_exp_editor = exp_editor.ExpressionEditor()
+
+    load_filter_button = Button('Load')
+    apply_filter_button = Button('Apply')
+    save_filter_button = Button('Save')
+    delete_filter_button = Button('Delete')
+    reset_filter_button = Button('Reset')
+    close_button = Text(locator='//div[@id="advsearchModal"]//button[@data-dismiss="modal"]')
+
+    save_filter_form = View.nested(AdvancedFilterSave)
+    load_filter_form = View.nested(AdvancedFilterLoad)
+    filter_user_input_form = View.nested(AdvancedFilterUserInput)
+
+
 class Search(View):
-    """ Represents search_text control
-    # TODO Add advanced search
-    """
-    search_text = Input(id="search_text")
-    search_btn = Text("//div[@id='searchbox']//div[contains(@class, 'form-group')]"
-                      "/*[self::a or (self::button and @type='submit')]")
-    clear_btn = Text(".//*[@id='searchbox']//div[contains(@class, 'clear')"
-                     "and not(contains(@style, 'display: none'))]/div/button")
+    """ Represents search_text control """
+    search_input = Input(id="search_text")
+    search_button = Text("//div[@id='searchbox']//div[contains(@class, 'form-group')]"
+                         "/*[self::a or (self::button and @type='submit')]")
+
+    clear_button = Text(".//*[@id='searchbox']//div[contains(@class, 'clear') "
+                        "and not(contains(@style, 'display: none'))]/div/button")
+    filter_clear_button = Text('//a[contains(@href, "adv_search_clear")]')
+    advanced_search_button = Button(title='Advanced Search')
 
-    def clear_search(self):
+    advanced_search_form = View.nested(AdvancedSearchView)
+
+# ================================= Simple Search ====================================
+
+    @property
+    def has_quick_search_box(self):
+        return self.search_input.is_displayed
+
+    def clear_simple_search(self):
+        """ Clear simple search field """
         if not self.is_empty:
-            self.clear_btn.click()
-            self.search_btn.click()
+            self.clear_button.click()
+            self.search_button.click()
 
-    def search(self, text):
-        self.search_text.fill(text)
-        self.search_btn.click()
+    def simple_search(self, text):
+        """ Search text using simple search """
+        self.search_input.fill(text)
+        self.search_button.click()
 
     @property
     @logged(log_result=True)
     def is_empty(self):
-        return not bool(self.search_text.value)
+        """ Checks if simple search field is emply """
+        return not bool(self.search_input.value)
+
+
+# ================================= Advanced Search ===============================
+    @property
+    def is_advanced_search_opened(self):
+        """Checks whether the advanced search box is currently opened"""
+        return self.advanced_search_form.search_exp_editor.is_displayed
```

Comment:
This is only a valid check for some advanced search modes, but doesn't cover when the save_filter_form or load_filter_form are displayed. the ExpressionEditor.ROOT locator isn't visible in these cases.

Suggest using multiple checks here against some of the nested views and ORing them to return.  That way if the save_form, load_form, etc are displayed instead of expression editor, this check still returns True.